### PR TITLE
[discovery] Enable discovery mode for SQL Server receiver

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

This changes enables automatic detection of SQL Server instances running in a host, docker, or kubernetes environment. The configuration is set up to automatically discover host and port, but requires username and password for connecting to SQL server. Also, the default metrics need to be updated to no longer match upstream.

**Testing:** <Describe what testing was performed and which tests were added.>
Tested in k8s env:
```
Discovering for next 10s...
Successfully discovered "sqlserver" using "k8s_observer" endpoint "k8s_observer/0ba5f848-8bbe-4008-8d61-3e46696f096e/(1433)".
Discovery complete.
```
Also pulled up OOTB dashboards, updated metrics to match OTEL naming, and they were populated properly with data from my k8s environment.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 💡 Enhancements 💡
+
+- (Splunk) `discovery`: Enable discovery mode for SQL Server receiver ([#](https://github.com/signalfx/splunk-otel-collector/pull/))
```
Comment (crobert-1) [inline CHANGELOG.md:7]:
```suggestion
- (Splunk) `discovery`: Enable discovery mode for SQL Server receiver ([#5109](https://github.com/signalfx/splunk-otel-collector/pull/5109))
```

File: cmd/otelcol/config/collector/config.d.linux/receivers/sqlserver.discovery.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#             enabled: false
+#         sqlserver.transaction_log.flush.wait.rate:
+#             enabled: false
+#         sqlserver.transaction_log.growth.count:
+#             enabled: false
+#         sqlserver.transaction_log.shrink.count:
+#             enabled: false
+#         sqlserver.transaction_log.usage:
+#             enabled: false
+#
```
Comment (crobert-1) [inline cmd/otelcol/config/collector/config.d.linux/receivers/sqlserver.discovery.yaml:81]:
```suggestion
# 
```
`make bundle.d` expects a whitespace here. We should fix it to not add an incorrect whitespace, but for now, we should just get the test to pass.

File: internal/confmapprovider/discovery/README.md [hunk 3]
```diff
@@ -144,6 +144,7 @@ I. Receivers
 * `smartagent` with `collectd/mysql` monitor type ([Linux](./bundle/bundle.d/receivers/smartagent-collectd-mysql.discovery.yaml))
 * `smartagent` with `collectd/nginx` monitor type ([Linux](./bundle/bundle.d/receivers/smartagent-collectd-nginx.discovery.yaml))
 * `smartagent` with `postgresql` monitor type ([Linux and Windows](./bundle/bundle.d/receivers/smartagent-postgresql.discovery.yaml))
+*  `sqlserver` ([Linux And Windows](./bundle/bundle.d/receivers/sqlserver.discovery.yaml))
```
Comment (jvoravong) [inline internal/confmapprovider/discovery/README.md:147]:
```suggestion
* `sqlserver` ([Linux And Windows](./bundle/bundle.d/receivers/sqlserver.discovery.yaml))
```

**Review by jvoravong — APPROVED** [review]

File: cmd/otelcol/config/collector/config.d.linux/receivers/sqlserver.discovery.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+#         message: The container is not serving http connections.
+#       - status: failed
+#         regexp: "unable to open tcp connection"
+#         message: Unable to resolve sql server tcp endpoint
+#       - status: partial
+#         regexp: "Login failed for user"
+#         message: >-
+#             Make sure your user credentials are correctly specified using the
+#             `--set splunk.discovery.receivers.sqlserver.config.username="<username>"` and
+#             `--set splunk.discovery.receivers.sqlserver.config.password="<password>"` command or the
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/config.d.linux/receivers/sqlserver.discovery.yaml:99]: Please leave only one way to configure done for other receivers: env vars.

**Comment by jvoravong** [comment]
Once all comments are addressed and the only tests failing are for TestExpandedDollarSignsViaStandardEnvVar, then feel free to merge. We are in the process of addressing these specific tests in a separate effort.

**Review by dmitryax — APPROVED** [review]
My comment can be addressed separately

**Pull request merged**